static/app/components/stacktracePreview.tsx

```diff
@@ -22,220 +22,208 @@ import getThreadStacktrace from './events/interfaces/threads/threadSelector/getT
-  }
-
-  renderHovercardBody(stacktrace: StacktraceType | undefined) {
-    const {event, loading, loadingVisible} = this.state;
-
-    if (loading && loadingVisible) {
-      return (
-        <NoStackTraceWrapper>
-          <LoadingIndicator hideMessage size={32} />
-        </NoStackTraceWrapper>
-      );
-    }
-
-    if (loading) {
-      return null;
-    }
-
-    if (!stacktrace) {
-      return (
-        <NoStackTraceWrapper onClick={this.handleStacktracePreviewClick}>
-          {t("There's no stack trace available for this issue.")}
-        </NoStackTraceWrapper>
-      );
-    }
-
-    const {organization, groupingCurrentLevel} = this.props;
+  // Not sure why we need to stop propagation, maybe to to prevent the hovercard from closing?
+  // If we are doing this often, maybe it should be part of the hovercard component.
+  const handleStacktracePreviewClick = React.useCallback((e: React.MouseEvent) => {
+    e.stopPropagation();
+  }, []);
 
+  const stacktrace = React.useMemo(() => {
     if (event) {
-      const platform = (event.platform ?? 'other') as PlatformType;
-
-      return (
-        <div onClick={this.handleStacktracePreviewClick}>
-          {!!organization.features?.includes('grouping-stacktrace-ui') ? (
-            <StackTraceContentV2
-              data={stacktrace}
-              expandFirstFrame={false}
-              includeSystemFrames={(stacktrace.frames ?? []).every(frame => !frame.inApp)}
-              platform={platform}
-              newestFirst={isStacktraceNewestFirst()}
-              event={event}
-              groupingCurrentLevel={groupingCurrentLevel}
-              isHoverPreviewed
-            />
-          ) : (
-            <StackTraceContent
-              data={stacktrace}
-              expandFirstFrame={false}
-              includeSystemFrames={(stacktrace.frames ?? []).every(frame => !frame.inApp)}
-              platform={platform}
-              newestFirst={isStacktraceNewestFirst()}
-              event={event}
-              isHoverPreviewed
-            />
-          )}
-        </div>
-      );
+      return getStacktrace(event);
     }
-
     return null;
-  }
-
-  render() {
-    const {children, disablePreview, theme, className} = this.props;
-
-    const {loading, loadingVisible} = this.state;
-    const stacktrace = this.getStacktrace();
-
-    if (disablePreview) {
-      return children;
-    }
-
-    return (
-      <span
-        className={className}
-        onMouseEnter={this.handleMouseEnter}
-        onMouseLeave={this.handleMouseLeave}
+  }, [event]);
+
+  const includeSystemFrames = React.useMemo(() => {
+    return stacktrace?.frames?.every(frame => !frame.inApp) ?? false;
+  }, [stacktrace]);
+
+  return (
+    <span
+      className={props.className}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      <StyledHovercard
+        body={
+          status === 'loading' && !loadingVisible ? null : status === 'loading' ? (
```

Comment:
Since this seems to be the meat of the component and the rest seems to be dealing with hovercard delays, I think it would probably be more correct if abstracted this into it's own component. This would slightly change the usage, but make the component more reusable and easier to test (no need to interact with hovercard)

That means that in our view components, we'd go from implicit hovercard usage to an explicit one
```
<StackTracePreview/>
```
to a more explicit one like
```
<Hovercard body={<StackTracePreview} />
```

(assuming we could get rid of the whole timeout/delay in favour of the hovercard delay prop)